mindsdb/integrations/handlers/byom_handler/byom_handler_exec.py

```diff
@@ -0,0 +1,343 @@
+            company_id=self.company_id,
+            integration_id=self.integration_id,
+        )
+
+        self.model_controller = WithKWArgsWrapper(
+            ModelController(),
+            company_id=self.company_id
+        )
+
+        self.parser = parse_sql
+        self.dialect = 'mindsdb'
+
+        self.is_connected = True
+
+    def get_tables(self) -> Response:
+        all_models = self.model_controller.get_models(integration_id=self.integration_id)
+        all_models_names = [[x['name']] for x in all_models]
+        response = Response(
+            RESPONSE_TYPE.TABLE,
+            pd.DataFrame(
+                all_models_names,
+                columns=['table_name']
+            )
+        )
+        return response
+
+    def get_columns(self, table_name: str) -> Response:
+        """ For getting standard info about a table. e.g. data types """  # noqa
+        predictor_record = get_model_record(company_id=self.company_id, name=table_name, ml_handler_name=self.name)
+        if predictor_record is None:
+            return Response(
+                RESPONSE_TYPE.ERROR,
+                error_message=f"Error: model '{table_name}' does not exists!"
+            )
+
+        data = []
+        if predictor_record.dtype_dict is not None:
+            for key, value in predictor_record.dtype_dict.items():
+                data.append((key, value))
+        result = Response(
+            RESPONSE_TYPE.TABLE,
+            pd.DataFrame(
+                data,
+                columns=['COLUMN_NAME', 'DATA_TYPE']
+            )
+        )
+        return result
+
+    def native_query(self, query: str) -> Response:
+        query_ast = self.parser(query, dialect=self.dialect)
+        return self.query(query_ast)
+
+    def query(self, query: ASTNode) -> Response:
+        statement = query
+
+        if type(statement) == Show:
+            if statement.category.lower() == 'tables':
+               return self.get_tables()
+            else:
+                response = Response(
+                    RESPONSE_TYPE.ERROR,
+                    error_message=f"Cant determine how to show '{statement.category}'"
+                )
+            return response
+        if type(statement) == CreatePredictor:
+            return self.learn(statement)
+        elif type(statement) == RetrainPredictor:
+            return self.retrain(statement)
+        elif type(statement) == DropPredictor:
+            return self.drop(statement)
+        elif type(statement) == Select:
+            model_name = statement.from_table.parts[-1]
+            where_data = get_where_data(statement.where)
+            predictions = self.predict(model_name, where_data)
+            return Response(
+                RESPONSE_TYPE.TABLE,
+                data_frame=pd.DataFrame(predictions)
+            )
+        else:
+            raise Exception(f"Query type {type(statement)} not supported")
+
+    def make_sql_session(self, company_id):
```

Comment:
This is now a common method in `integrations/utils`

Reply:
done